### ISSUE TITLE
Weird issues with SNES Controller

### ISSUE THREAD

**Issue by 7ngel** [mgba-emu/mgba#3120]
So I'm having an issue with my bluetooth official nintendo SNES controller (plugged in or through blutooth) It detects the controller and all the correct key presses on the controller while binding them in mGBA but when I got to use the controller on a game like wario land 4 the controller is constantly going left and I can't do anything about it. I've tested my xbox series X controller and it works completely flawlessly, and I've tested the SNES controller on other emulators like dolphin and it works fine there. Any solutions?

VERSION 0.10.3 64 BIT WINDOWS 10

**Comment by endrift** [comment]
Can you please attach the zip file generated by File > Report Bug?

**Comment by bionic-bill** [comment]
[snes-controller-bug.zip](https://github.com/mgba-emu/mgba/files/14077012/snes-controller-bug.zip)

Hello! I have the same issue. In 0.10.2 I could get it to work by closing Steam which made the controller show up in the menu as "Wireless Gamepad" . With steam open it shows up as "Nintendo SNES Controller" . In 0.10.3 it shows up with the correct name but it doesn't work even with steam closed. I have attached the bug report. Thanks!

**Comment by 7ngel** [comment]
> Can you please attach the zip file generated by File > Report Bug?

[bug.zip](https://github.com/mgba-emu/mgba/files/14078615/bug.zip)

Have you seen the guys message above aswell seem like this could be a cause of a new update.

**Comment by endrift** [comment]
I've obtained an NSO SNES Controller and will test this soon. I had assumed something strange in the config file might be responsible but I can't find anything.

**Comment by 7ngel** [comment]
> I've obtained an NSO SNES Controller and will test this soon. I had assumed something strange in the config file might be responsible but I can't find anything.

Were you able to test it?

**Comment by endrift** [comment]
Yes, the issue seems to be rooted in SDL. Click on each of the buttons that has a - or + on it and hit "clear analog". This should fix it. I'll need to look into how to avoid this issue better in the future.

**Comment by bionic-bill** [comment]
Thanks endrift, that fixed it! I wish i had been a little more daring (or even noticed the button 😅) messing with the settings.
